src/pudl_archiver/depositors/zenodo.py

```diff
@@ -219,46 +222,40 @@ async def get_new_version(
             else:
                 return deposition
 
-        url = f"{self.api_root}/deposit/depositions/{deposition.id_}/actions/newversion"
+        new_deposition = await self.get_new_deposition_version(deposition)
+        # If files already in new deposition, remove.
+        if len(new_deposition.files) > 0:
+            await self.delete_deposition(new_deposition)  # DEBUG ME!
+            new_deposition = await self.get_new_deposition_version(
+                deposition
+            )  # Debug: Something isn't working here!
 
-        # create a new unpublished deposition version
+        # Link files from previous deposition.
+        await self.link_previous_files(new_deposition)
+
+        return new_deposition
+
+    async def get_new_deposition_version(self, deposition: Deposition):
+        """Create a new draft deposition version."""
+        url = f"{self.api_root}/records/{deposition.id_}/versions"
         response = await self.request(
             "POST",
             url,
-            log_label="Creating new version",
+            log_label="Creating new version.",
             headers=self.auth_write,
         )
-        old_deposition = Deposition(**response)
-        new_deposition_url = old_deposition.links.latest_draft
-
-        source_metadata = old_deposition.metadata.dict(by_alias=True)
-        metadata = {}
-        for key, val in source_metadata.items():
-            if key not in ["doi", "prereserve_doi", "publication_date"]:
-                metadata[key] = val
-
-        previous = semantic_version.Version(source_metadata["version"])
-        version_info = previous.next_major()
-
-        metadata["version"] = str(version_info)
-
-        # Update metadata of new deposition with new version info
-        data = json.dumps({"metadata": metadata})
-
```

Comment:
Add this whole section back as these are removed by the `create a new version` endpoint.